zebra-state/src/service/memory_state.rs

```diff
@@ -480,16 +482,18 @@ impl NonFinalizedState {
             })
             .expect("commit_block is only called with blocks that are ready to be commited");
 
-        parent_chain.push(block);
+        parent_chain.push(block.clone());
         self.chain_set.insert(parent_chain);
+        self.update_metrics_for_committed_block(block);
```

Comment:
```suggestion
        self.update_metrics_for_committed_block(&block);
        parent_chain.push(block);
        self.chain_set.insert(parent_chain);
```

Reply:
@yaahc what's the benefit of this change?

The original code updates the metrics last, so that the chain metrics are accurate.
If we apply your suggestion, then it updates the metrics first, then changes the chains.
So the metrics will always be out of date.

Should I add a comment that explains the order?

Reply:
>  left some suggestions to avoid a clone that I think is unnecessary

Oops didn't see the review comment.